From 4f5f51f196afc25bd3e3251a43b7eca46e7160d5 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 23:11:07 -0400
Subject: [ethernet] cxgb4vf: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <1410477072-95160-3-git-send-email-svemuri@redhat.com>
Patchwork-id: 93901
O-Subject: [RHEL7.1 PATCH BZ 1070945 2/7] cxgb4vf: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1070945
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range() and pci_enable_msix_range()
interfaces.

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Cc: Casey Leedom <leedom@chelsio.com>
Cc: netdev@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit bd66368940ae1c8c7491f55650b813c9d790f92c)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c b/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c
index 509fed9..1d0fe9b 100644
--- a/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf_main.c
@@ -2444,7 +2444,7 @@ static void reduce_ethqs(struct adapter *adapter, int n)
  */
 static int enable_msix(struct adapter *adapter)
 {
- int i, err, want, need;
+ int i, want, need, nqsets;
  struct msix_entry entries[MSIX_ENTRIES];
  struct sge *s = &adapter->sge;
 
@@ -2460,25 +2460,23 @@ static int enable_msix(struct adapter *adapter)
   */
  want = s->max_ethqsets + MSIX_EXTRAS;
  need = adapter->params.nports + MSIX_EXTRAS;
- while ((err = pci_enable_msix(adapter->pdev, entries, want)) >= need)
-  want = err;
 
- if (err == 0) {
-  int nqsets = want - MSIX_EXTRAS;
-  if (nqsets < s->max_ethqsets) {
-   dev_warn(adapter->pdev_dev, "only enough MSI-X vectors"
-     " for %d Queue Sets\n", nqsets);
-   s->max_ethqsets = nqsets;
-   if (nqsets < s->ethqsets)
-    reduce_ethqs(adapter, nqsets);
-  }
-  for (i = 0; i < want; ++i)
-   adapter->msix_info[i].vec = entries[i].vector;
- } else if (err > 0) {
-  dev_info(adapter->pdev_dev, "only %d MSI-X vectors left,"
-    " not using MSI-X\n", err);
+ want = pci_enable_msix_range(adapter->pdev, entries, need, want);
+ if (want < 0)
+  return want;
+
+ nqsets = want - MSIX_EXTRAS;
+ if (nqsets < s->max_ethqsets) {
+  dev_warn(adapter->pdev_dev, "only enough MSI-X vectors"
+    " for %d Queue Sets\n", nqsets);
+  s->max_ethqsets = nqsets;
+  if (nqsets < s->ethqsets)
+   reduce_ethqs(adapter, nqsets);
  }
- return err;
+ for (i = 0; i < want; ++i)
+  adapter->msix_info[i].vec = entries[i].vector;
+
+ return 0;
 }
 
 static const struct net_device_ops cxgb4vf_netdev_ops = {
-- 
1.7.1